File: test_fixtures/security_vuln/sql_injection.java
// SQL injection variants in Java.
// Purpose: verify Java-specific SQLi patterns are detected.
package com.example.dao;

import java.sql.*;

public class OrderDAO {

    private Connection conn;

    public OrderDAO(Connection conn) {
        this.conn = conn;
    }

    public ResultSet findOrders(String customerName) throws SQLException {
        // BUG: SQL injection via string concatenation
        String sql = "SELECT * FROM orders WHERE customer = '" + customerName + "'";
        Statement stmt = conn.createStatement();
        return stmt.executeQuery(sql);
    }

    public int updateOrder(String orderId, String status) throws SQLException {
        // BUG: SQL injection in UPDATE
        String sql = "UPDATE orders SET status = '" + status
                   + "' WHERE id = '" + orderId + "'";
        Statement stmt = conn.createStatement();
        return stmt.executeUpdate(sql);
    }

    public void deleteProduct(String productId) throws SQLException {
        // BUG: SQL injection in DELETE
        Statement stmt = conn.createStatement();
        stmt.execute("DELETE FROM products WHERE id = '" + productId + "'");
    }
}

File: test_fixtures/security_vuln/sql_injection.go
// SQL injection variants in Go.
// Purpose: verify Go-specific SQLi patterns are detected.
package data

import (
	"database/sql"
	"fmt"
)

func QueryUserConcat(db *sql.DB, userID string) error {
	// BUG: SQL injection via string concatenation
	query := "SELECT * FROM users WHERE id = '" + userID + "'"
	_, err := db.Query(query)
	return err
}

func QueryUserSprintf(db *sql.DB, name string) error {
	// BUG: SQL injection via fmt.Sprintf
	query := fmt.Sprintf("SELECT * FROM users WHERE name = '%s'", name)
	_, err := db.Query(query)
	return err
}

func DeleteOrderSprintf(db *sql.DB, orderID string) error {
	// BUG: SQL injection in DELETE with Sprintf
	q := fmt.Sprintf("DELETE FROM orders WHERE id = '%s'", orderID)
	_, err := db.Exec(q)
	return err
}
